legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/serialization/filesystem/PureCodeStorage.java

```diff
@@ -442,6 +443,22 @@ public boolean isVersioned(String path)
         return (codeStorage instanceof VersionControlledCodeStorage) && ((VersionControlledCodeStorage) codeStorage).isVersioned(path);
     }
 
+    @Override
+    public boolean isRepositoryImmutable(String name)
+    {
+        CodeRepository repo = getRepository(name);
+        if (repo == null)
+        {
+            return false;
+        }
+        RepositoryCodeStorage repositoryCodeStorage = this.codeStorages.select(storage -> storage.getRepositories().contains(repo)).getFirst();
+        if (repositoryCodeStorage == null)
+        {
+            return false;
+        }
+        return repositoryCodeStorage instanceof ImmutableRepositoryCodeStorage;
```

Comment:
Repository code storages are already indexed by repository name. Also, the null check is redundant, since the instanceof check will return false for null.
```suggestion
        return this.codeStorageByName.get(name) instanceof ImmutableRepositoryCodeStorage;
```